specifyweb/frontend/js_src/lib/components/QueryBuilder/Wrapped.tsx

```diff
@@ -145,7 +145,8 @@ export function QueryBuilder({
 
   const isResourceModified = useIsModified(queryResource);
   const saveRequired =
-    (isResourceModified || state.saveRequired) && !isEmbedded;
+    isResourceModified || (state.saveRequired && !isEmbedded);
```

Comment:
A better variable naming + reuses the variable instead of stating logic twice:

```tsx
   const saveRequired =
      isResourceModified || state.saveRequired;
    const promptToSave = saveRequired && !isEmbeeded;
```

And then use `promptToSave` for useUnloadProtect, and use `saveRequired` everywhere else

Reply:
isn't prompToSave everywhere else to keep the same behavior as before?

Reply:
The only other usage of `saveRequired` is in QueryHeader to determine if the save button should be enabled or not

But QueryHeader is not rendered when isEmbedded anyway so it doesn't make a difference.

And in the future, once we do show enable QueryHeader for embedded queries, using saveRequired instead of promtToSave will result in behavior were user won't be required to save, but will be able to save if they want to - desirable behavior.